### ISSUE TITLE
Linux Telegram app freezes on message

### ISSUE THREAD

**Issue by Alex-Blade** [telegramdesktop/tdesktop#8768]
### Steps to reproduce
1. Install Telegram Desktop on Linux (tested on [Fedora/Pop/Manjaro] + Gnome 3)
2. Turn on notifications sounds (Turned on by default)
3. Enjoy frozen application whenever new message comes (freezes till the notifications appears)

### Expected behaviour
Telegram should continue working

### Actual behaviour
The app freezes (even if the message is already in chat list) and continues working when the notification appears

### Configuration
**Operating system:** Any Linux distro with latest Gnome 3
**Audio solution:** Creative Sound Blaster Z, internal audio

**Version of Telegram Desktop: 2.4.2**

**Installation source (Linux Only)** - the official website + GitHub releases + flatpak (tested all)

**Used theme**: Dark mode

**Comment by ilya-fedin** [comment]
Is this reproducible with with X11 or only with Wayland?

**Comment by krims0n32** [comment]
I have the same issue, Archlinux using Awesome as window manager (X11). This started happening in Telegram desktop 2.3+. 2.2 works fine.

It only happens when using native notifications and for me it also happens when notification sounds are turned off.

**Comment by ilya-fedin** [comment]
I can't reproduce, native notifications are working just fine for me

**Comment by GustavoKatel** [comment]
I have the same issue :(

Telegram 2.3.2 (manjaro repo)
Manjaro + awesomewm + x11

**Comment by cyberpunkrocker-zero** [comment]
The same issue here: Awesome WM on Arch Linux / x11, currently telegram-desktop- 2.4.0-1. The issue started after upgrading to 2.3.x
When I get a message, the red number appears to systray icon, the sound notification sounds, and then telegram freezes completely for 10-20 seconds, after telegram is revived the actual notification appears.

**Comment by kierun** [comment]
Same issue: 

```
√ ; rpm -qa | rg telegram-desktop
telegram-desktop-2.3.2-1.fc32.x86_64
√ ; uname -a
Linux nightwatch.neverness.org 5.8.12-200.fc32.x86_64 #1 SMP Mon Sep 28 12:17:31 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
√ ; awesome --version
awesome v4.3 (Too long)
 • Compiled against Lua 5.3.5 (running with Lua 5.3)
 • D-Bus support: ✔
 • execinfo support: ✔
 • xcb-randr version: 1.6
 • LGI version: 0.9.2
√ ; rpm -qa | rg X11
libX11-devel-1.6.12-1.fc32.x86_64
libX11-xcb-1.6.12-1.fc32.x86_64
libX11-common-1.6.12-1.fc32.noarch
libX11-1.6.12-1.fc32.x86_64
```

**Comment by ilya-fedin** [comment]
It's been a long time since 2.3.2, why there are a lot of people here now?

**Comment by cyberpunkrocker-zero** [comment]
A Reddit-effect? Me, I was about to report this over a week ago, but never did. Now people are asking about it in Reddit, so...

**Comment by ilya-fedin** [comment]
> A Reddit-effect?

Can you give a link?

**Comment by kierun** [comment]
> It's been a long time since 2.3.x, why there are a lot of people here now?

Because that's what is in the package manager on Fedora and I saw a reddit post on it in `/r/awesome`…

**Comment by kierun** [comment]
[Here is the reddit post.](https://www.reddit.com/r/awesomewm/comments/j6nzky/telegram_notification_freeze/)

**Comment by GustavoKatel** [comment]
> > It's been a long time since 2.3.x, why there are a lot of people here now?
> 
> Because that's what is in the package manager on Fedora and I saw a reddit post on it in `/r/awesome`…

same for manjaro

**Comment by ilya-fedin** [comment]
Is awesome has its own notification daemon? Is this reproducible with any other notification daemon?

**Comment by GustavoKatel** [comment]
> Does awesome has its own notification daemon? Is this reproducible with any other notification daemon?

yes, awesome has its own notification daemon, but the OP mentioned they're using gnome 3?

**Comment by ilya-fedin** [comment]
> but the OP mentioned they're using gnome 3?

Ok, any other than awesome and gnome.

**Comment by kierun** [comment]
I am afraid that the bug is still appearing on 2.4.2 with native notifications: No notifications are shown and the TG window becomes unresponsive (frozen) for a dozen seconds.

**Comment by ilya-fedin** [comment]
> I am afraid that the bug is still appearing on 2.4.2 with native notifications

read previous messages

**Comment by fuzxi** [comment]
> > but the OP mentioned they're using gnome 3?
> 
> Ok, is this reproducible with any other notification daemon rather than awesome and gnome?

I can reproduce this using mako on sway.

**Comment by ilya-fedin** [comment]
Oh, that's bad, we still don't know what cause this. And I can't reproduce, I don't experience any freezes.

**Comment by krims0n32** [comment]
I have tested using Dunst (https://dunst-project.org/) instead of Awesome's notification manager and that does not give me any freezes. So it seems to be a combination of Telegram and notification daemon. I'm not sure what Gnome uses for notifications but apparently it is not an Awesome specific issue.

Has something changed in the content of the notification Telegram sends, perhaps the profile picture or something that borks out on some notification daemons?

**Comment by ilya-fedin** [comment]
> Has something changed in the content of the notification Telegram sends

dbus library that Telegram uses to send a notification

**Comment by Alex-Blade** [comment]
> Is this reproducible with with X11 or only with Wayland?
Right now I'm on:
Fedora 32 (Workstation Edition) with GNOME Shell 3.36.6, X11. I'm not sure about Wayland though

**Comment by Alex-Blade** [comment]
How do I check what notification manager do I use? Didn't change anything, so I guess it's default Gnome thing

**Comment by ilya-fedin** [comment]
Tried to use a more low-leverl. Can anyone check if this fixes the issue?
https://github.com/ilya-fedin/tdesktop/actions/runs/296715634

**Comment by krims0n32** [comment]
This seems to fix the issue for me!

**Comment by Alex-Blade** [comment]
`QCoreApplication::postEvent: Unexpected null receiver
`

**Comment by ilya-fedin** [comment]
Since opinions were divided, more testers are needed.

> Tried all three versions

Only just "Telegram" is need to try

**Comment by cyberpunkrocker-zero** [comment]
Just got the patched version compiled and tested; so far, the patch seems to work as intended, no more freezes or any other oddities, desktop notifications are on.
Patch applied to 2.4.3 on Arch Linux / Awesome wm / x11

**Comment by ilya-fedin** [comment]
Maybe @Alex-Blade's issue is another after all? :thinking:

**Comment by ilya-fedin** [comment]
@Alex-Blade can you check if downgrading to 2.2 helps for you like for others?

**Comment by Alex-Blade** [comment]
@ilya-fedin, that's extremely weird, I've been using Telegram Flathub version for the whole day and only now understood that there have been no freezes... All the versions you've provided are also working correctly. Will continue to observe and investigate, hope the issue is resolved

**Comment by ilya-fedin** [comment]
> that's extremely weird, I've been using Telegram Flathub version for the whole day and only now understood that there have been no freezes...

Maybe the issue was on GNOME side and you updated the system?

**Comment by Alex-Blade** [comment]
Maybe it was on hardware optimization side at all, I'm actually not really satisfied with stability of Ryzen + NVMe combo on Linux (needed to solve problems for both separately and then combine). Didn't notice any updates, but possible, I regretfully don't have a log of updates

**Comment by cyberpunkrocker-zero** [comment]
Seems like it
